### PR TITLE
SimpleMenu: Allow elments to be passed as prop-types

### PR DESCRIPTION
**Pull request by gu-stav**
In https://github.com/strapi/strapi/pull/12213/files#diff-945b0ae1436ead2b4ac56d7921a076ac866efcd77ed5985466444d07b1751d78R30-R56 I'd like to create a custom label for the `SimpleMenu`. It works already, by the prop-types don't allow elements yet. This PR adds `element`s next to `string` and `number` as valid types.

An example was added too.

Refs: https://github.com/strapi/strapi/pull/12213

### CODE REVIEW

**Comment by HichamELBSI** [comment]
This modification will break the way we define the default item to focus in the list.

```js
  // SimpleMenu.js 
  useEffect(() => {
    // Useful to focus the selected item in the list
    const defaultItemIndexToFocus = childrenArray.findIndex((c) => c.props.children === label);
    if (defaultItemIndexToFocus !== -1) {
      setFocusItem(defaultItemIndexToFocus);
    }
  }, [label]);
```

File: packages/strapi-design-system/src/SimpleMenu/SimpleMenu.js [hunk 1]
```diff
@@ -97,10 +97,18 @@ export const SimpleMenu = ({ label, children, id, as: asComp, ...props }) => {
...
-    if (defaultItemIndexToFocus !== -1) {
-      setFocusItem(defaultItemIndexToFocus);
+    if (['string', 'number'].includes(typeof label)) {
+      // Useful to focus the selected item in the list
+      const defaultItemIndexToFocus = childrenArray.findIndex((c) => c.props.children === label);
+
+      if (defaultItemIndexToFocus !== -1) {
+        setFocusItem(defaultItemIndexToFocus);
+      }
+    } else if (typeof label === 'object') {
```
Comment (gu-stav) [inline packages/strapi-design-system/src/SimpleMenu/SimpleMenu.js:107]: @HichamELBSI I'm not sure about the check, but what do you think about this? I feel like this works properly in all cases.

**Comment by HichamELBSI** [comment]
@gu-stav Can you just add a condition to this useEffect to check if label is not an element so we can merge this one 👍🏽

**Review by HichamELBSI — APPROVED** [review]
LGTM 👍🏽

File: packages/strapi-design-system/src/SimpleMenu/SimpleMenu.js (hunk 1 above)
Comment (soupette) [reply packages/strapi-design-system/src/SimpleMenu/SimpleMenu.js:107]: Maybe you can check if the object as `$$typeof` key?

Comment (HichamELBSI) [reply packages/strapi-design-system/src/SimpleMenu/SimpleMenu.js]: Maybe you can use `React.isValidElement(label)` instead?

**Pull request merged**